src/esi/Parser.cc

```diff
@@ -25,19 +27,29 @@ ESIParser::GetRegistry()
 ESIParser::Pointer
 ESIParser::NewParser(ESIParserClient *aClient)
 {
-    if (Parser == NULL) {
-        Parser = GetRegistry().front();
-
-        // if type name matters, use it
+    if (!Parser) {
+        const char *use;
         if (strcasecmp(Type, "auto") != 0) {
-            for (auto *p : GetRegistry()) {
-                if (p && strcasecmp(p->name, Type) != 0)
-                    Parser = p;
```

Comment:
Hm.. So when esi_parser was explicitly set to, say, "libxml2", this official code was trying to find a parser _not_ named "libxml2"?! Another bug (fixed by this PR)? If so, please adjust the PR description to mention that this bug was also fixed.

Reply:
Good spotting. Done.